src/Bicep.Core/FileSystem/InMemoryFileResolver.cs

```diff
@@ -57,5 +79,30 @@ public IEnumerable<Uri> GetFiles(Uri fileUri, string pattern)
         {
             return fileLookup.Keys;
         }
+
+        public bool TryReadAsBase64(Uri fileUri, [NotNullWhen(true)] out string? fileBase64, [NotNullWhen(false)] out DiagnosticBuilder.ErrorBuilderDelegate? failureBuilder, int maxCharacters = -1)
+        {
+            if (!fileLookup.TryGetValue(fileUri, out var fileContents))
+            {
+                failureBuilder = x => x.ErrorOccurredReadingFile(missingFileFailureBuilder(fileUri));
+                fileBase64 = null;
+                return false;
+            }
+
+            failureBuilder = null;
+            var bytes = Encoding.UTF8.GetBytes(fileContents);
+            if (maxCharacters > 0)
+            {
+                var maxBytes = maxCharacters / 4 * 3;
```

Comment:
Could you add a comment explaining this calculation (each base64 char represents 6 bits)

Reply:
done.

Reply:
added